src/solutions/ode_solutions.jl

```diff
@@ -15,7 +15,40 @@ struct ODESolution{T,N,uType,uType2,DType,tType,rateType,P,A,IType,DE} <: Abstra
   destats::DE
   retcode::Symbol
 end
-(sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+
+function (sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left)
+  if deriv != Val{0}
```

Comment:
Why do we need to restrict it to `Val{0}`? Isn't this just passed through to the calls of `sol.interp`?

Reply:
AFAIK, higher-order interpolation is currently not supported yet. https://github.com/SciML/ModelingToolkit.jl/issues/1014 This is just an effort to to indicate to the user that higher-order interpolation is not implemented yet. @YingboMa suggested that we could implement a ForwardDiff based higher-order interpolation -- but it might warrant a separate PR as it doesn't seem straightforward.

Reply:
I guess I will make it more clear with -- "Higher-order interpolation is not implemented."

Reply:
IMO it is more correct and easier to fix if it fails when calling `sol.interp(....)` in case it is missing. Alternatively, if one really wants to exclude this case on a higher level, it would be more idiomatic to just restrict the function signature to `Val{0}` instead of throwing a custom error.